lib/purge-calls.js

```diff
@@ -8,6 +8,11 @@ async function purgeCalls(client, logger) {
   logger = logger || noopLogger;
   let purgedCount = 0;
   try {
+    if (await client.getAsync(PURGE_CALLS_LOCK_KEY)) {
```

Comment:
a better way to do this -- set a key if it has not been set -- is to use the SET command with the NX option.

https://redis.io/commands/set/

this would eliminate the race condition in your code.

Reply:
Makes sense. Fix pushed